examples/gno.land/p/demo/accesscontrol/accesscontrol_test.gno

```diff
@@ -0,0 +1,228 @@
+package accesscontrol
+
+import (
+	"std"
+	"testing"
+
+	"gno.land/p/demo/testutils"
+	"gno.land/p/demo/uassert"
+)
+
+// TestAccessControl verifies the access control functionality.
+func TestAccessControl(t *testing.T) {
+	admin := testutils.TestAddress("admin")
+	user1 := testutils.TestAddress("user1")
+	user2 := testutils.TestAddress("user2")
+
+	// Create new RoleData
+	roleData := NewRole("admin", admin)
+
+	// Check initial admin role
+	if roleData.AdminRole != admin {
+		t.Fatalf("expected admin role to be %s, got %s", admin.String(), roleData.AdminRole.String())
+	}
+	!uassert.Equal(t, roleData.AdminRole, admin)
+
+	// Grant role to user1
+	std.TestSetOrigCaller(admin)
+	roleData.GrantRole(user1)
+	if !roleData.HasRole(user1) {
+		t.Fatalf("expected user1 to have role")
+	}
+
+	// Check that user2 does not have the role
+	if roleData.HasRole(user2) {
+		t.Fatalf("expected user2 not to have role")
+	}
+
+	// Revoke role from user1
+	roleData.RevokeRole(user1)
+	if roleData.HasRole(user1) {
+		t.Fatalf("expected user1 not to have role after revocation")
+	}
+
+	// Grant role to user1 again
+	roleData.GrantRole(user1)
+
+	// User1 renounces the role
+	std.TestSetOrigCaller(user1)
+	roleData.RenounceRole(user1)
+	if roleData.HasRole(user1) {
+		t.Fatalf("expected user1 not to have role after renouncing")
+	}
+
+	// Change admin role to user2
+	std.TestSetOrigCaller(admin)
+	roleData.SetRoleAdmin(user2)
+	if roleData.AdminRole != user2 {
+		t.Fatalf("expected admin role to be %s, got %s", user2.String(), roleData.AdminRole.String())
+	}
+
+	// User1 (now not admin) tries to grant role to user2, should panic
+	std.TestSetOrigCaller(user1)
+	defer func() {
+		if r := recover(); r == nil {
+			t.Fatalf("expected panic when non-admin tries to grant role")
+		}
+	}()
+	roleData.GrantRole(user2)
```

Comment:
If the package methods didn't panic, you wouldn't need to recover like this in a test